core-api/src/main/java/com/redhat/lightblue/Response.java

```diff
@@ -101,6 +135,37 @@ public void addEntityData(JsonNode doc) {
     }
 
     /**
+     * Ensures that the response is not larger than maxResultSetSizeB to protect Lightblue from running out of memory when building large result sets.
+     *
+     * This method is used to checks only the data part of the response. TODO: check dataErrors as well, as they include entire docs.
+     *
+     * @param entityDataJson json being added to the response
+     */
+    private void enforceResponseSizeLimits(JsonNode entityDataJson) {
+
+        if (isCheckResponseSize()) {
+            responseDataSizeB += JsonUtils.size(entityDataJson);
+
+            // TODO: could account for copies made for hooks here
```

Comment:
What do you think about multiplying the size to account for the copies made for hooks (the number of copies depends on the request kind)? Consider this a workaround until the real solution: https://github.com/lightblue-platform/lightblue-mongo/issues/378